pydy/viz/scene.py

```diff
@@ -459,33 +458,59 @@ def remove_static_html(self, force=False):
             print "aborted!"
 
     def display(self):
-        """
-        display method can be used in two ways.
-        When called from IPython notebook, it shows the visualization
-        in the form of output cell in the IPython notebook.
-        If it is called from python interpreter or
-        IPython interpreter(not notebook), It generates an html file,
-        in the current directory, which can be opened in the webgl
-        compliant browser for viewing the visualizations.
-
-        The simulation data is used from this scene, hence
-        all simulation data generation methods should be called before
-        calling this method
-
-        """
+        """Displays the scene in the default webbrowser."""
         self.create_static_html()
-        run_server(scene_file=self.scene_json_file)
+        run_server(scene_file=self._scene_json_file)
+
+    def _rerun_button_callback(self, clicked):
+        """Callback for the "Rerun Simulation" button. When executed the
+        parameter values are collected from the text input widgets and used
+        in a new simulation of the model."""
+
+        if ipython_less_than_3:
+            self._rerun_button.add_class('disabled')
+        else:
+            self._rerun_button._dom_classes = ['btn-info', 'active', 'disabled']
+
+        self._rerun_button.description = 'Rerunning Simulation...'
+
+        original_scene_file = self._scene_json_file
+        try:
+            self._system.constants = {s: w.value for s, w in
+                                      self._constants_text_widgets.items()}
+            self.generate_visualization_json_system(self._system)
+        except:
```

Comment:
It is impossible to predict what the error would be. There are too many possibilities. I just want it to prevent hanging the callback if there is any error at all.